x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolver.java

```diff
@@ -182,35 +210,52 @@ private Tuple<ResolvedEnrichPolicy, String> mergeLookupResults(
                 } else {
                     final String failure = lookupResult.failures.get(policyName);
                     if (failure != null) {
-                        failures.add(failure);
+                        if (skipUnavailable) {
+                            remotesToBeSkipped.put(cluster, new IllegalStateException(failure));
+                        } else {
+                            failures.add(failure);
+                        }
+                    } else if (skipUnavailable) {
+                        // code path where there was no enrich policy found at all on the remote
+                        remotesToBeSkipped.put(cluster, new IllegalStateException("failed to resolve enrich policy [" + policyName + "]"));
                     }
                 }
             }
         }
-        if (targetClusters.size() != policies.size()) {
+        if (targetClusters.size() != policies.size() + remotesToBeSkipped.size()) {
             final String reason;
             if (failures.isEmpty()) {
-                List<String> missingClusters = targetClusters.stream().filter(c -> policies.containsKey(c) == false).sorted().toList();
-                reason = missingPolicyError(policyName, targetClusters, missingClusters);
+                List<String> missingClusters = targetClusters.keySet()
+                    .stream()
+                    .filter(c -> policies.containsKey(c) == false)
+                    .sorted()
+                    .toList();
+                reason = missingPolicyError(policyName, targetClusters.keySet(), missingClusters);
             } else {
                 reason = "failed to resolve enrich policy [" + policyName + "]; reason " + failures;
             }
-            return Tuple.tuple(null, reason);
+            return new MergedPolicyLookupResult(null, remotesToBeSkipped, reason);
+        } else if (policies.isEmpty()) {
+            // no target cluster had a valid enrich policy and all are skip_unavailable=true
+            assert targetClusters.values().stream().allMatch(b -> b == Boolean.TRUE) : "Not all target clusters are skip_unavailable=true";
+            return new MergedPolicyLookupResult(null, remotesToBeSkipped, null);
         }
+
         Map<String, EsField> mappings = new HashMap<>();
         Map<String, String> concreteIndices = new HashMap<>();
         ResolvedEnrichPolicy last = null;
+        // loop over clusters with a ResolvedEnrichPolicy - ensure no errors within the policy
```

Comment:
Note that mismatches _across_ policies (being checked in the section below here) are still fatal errors for skip_unavailable=true clusters. I started down the road of having these be skippable errors, but that looks rather tricky to pull off. At a minimum, you'd have to partition the policies by "skip_unavailable" and build a canonical list of fields/types/etc. from skip_un=false clusters and then compare the skip_un=true clusters and then if any mismatches are found, those aren't fatal, but you pull that cluster out of the list to be resolved for field-caps. Not impossible but this section would require a significant rewrite, so I decided to only handle missing enrich policies (and policies that have errors on the remote cluster during resolution), but not fail them based on mismatches between policies.